packages/shared/src/utils.ts

```diff
@@ -211,3 +217,15 @@ export const fileSizeToBytes = (input: string) => {
   const bytes = sizeValue * Math.pow(1024, FILESIZE_UNITS.indexOf(sizeUnit));
   return Math.floor(bytes);
 };
+
+export async function safeParseJSON<T>(
+  input: string | { text: () => Promise<string> },
+): Promise<T | Error> {
+  const text = typeof input === "object" ? await input.text() : input;
+  try {
+    return JSON.parse(text) as T;
```

Comment:
response/request.text() then JSON.parse might not be the best here, but first awaiting .json() before falling back to awaiting .text() doesn't feel right either...

Reply:
I'll think on this, while it may not be the best, I feel like it's still a pretty reasonable way to handle this, even if we lose a little bit of efficiency.

Reply:
I guess the clone method you used in #415 is a little cleaner